### PR TITLE
Add udpated ebuild so that the package supports current gcc

### PR DESCRIPTION
**Pull request by dr-diem**
Also fixes bugs 669212 & 703090.

### CODE REVIEW

**Comment by dr-diem** [comment]
Closes: https://bugs.gentoo.org/724882
Closes: https://bugs.gentoo.org/724884
Closes: https://bugs.gentoo.org/724886

File: games-action/descent1-freedata/descent1-freedata-1.ebuild [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+LICENSE="public-domain"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+# If all USE flags are unset, this ebuild installs zero files.  Require
+# at least one to be set.
+REQUIRED_USE="|| ( ${IUSE//+/} )"
+
+DEPEND=""
+RDEPEND=""
```
[the remark this answers is not in the document]
Comment (dr-diem) [reply games-action/descent1-freedata/descent1-freedata-1.ebuild:38]: Gotcha - I'll leave out the DEPEND and add the RDEPEND as you suggest to block installation for 0.59.100 and lower.

File: games-action/descent1-freedata/descent1-freedata-1.ebuild [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
```
[the remark this answers is not in the document]
Comment (dr-diem) [reply games-action/descent1-freedata/descent1-freedata-1.ebuild:39]: I don't know - is it related to the current EAPI=6? Regardless, I will add the line you indicate.

File: games-action/dxx-rebirth/metadata.xml [hunk 3]
```diff
@@ -6,8 +6,8 @@
 		<name>Gentoo Games Project</name>
 	</maintainer>
 	<maintainer type="person">
-		<email>totktonada.ru@gmail.com</email>
-		<name>Alexander Turenko</name>
+		<email>iansilvester@fastmail.fm</email>
+		<name>Ian Silvester</name>
```
[the remark this answers is not in the document]
Comment (dr-diem) [reply games-action/dxx-rebirth/metadata.xml:10]: Okay yes, will do.

File: games-action/descent1-freedata/descent1-freedata-1.ebuild (hunk 2 above)
[the remark this answers is not in the document]
Comment (chewi) [reply games-action/descent1-freedata/descent1-freedata-1.ebuild:39]: I don't think so. `S` defaults to `${WORKDIR}/${P}`, which won't exist in this case.

File: games-action/dxx-rebirth/dxx-rebirth-0.61.0_pre20190720.ebuild [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+
+# This file is part of the DXX-Rebirth project.
+#
+# It is copyright by its individual contributors, as recorded in the
+# project's Git history.  See COPYING.txt at the top level for license
+# terms and a link to the Git history.
+
+EAPI=6
+
+inherit eutils scons-utils toolchain-funcs xdg
```
[the remark this answers is not in the document]
Comment (dr-diem) [reply games-action/dxx-rebirth/dxx-rebirth-0.61.0_pre20190720.ebuild:12]: This is being applied upstream, though he (and I) have only 3.7 to test against for now.

File: games-action/descent1-freedata/descent1-freedata-1.ebuild (hunk 2 above)
[the remark this answers is not in the document]
Comment (dr-diem) [reply games-action/descent1-freedata/descent1-freedata-1.ebuild:39]: This has now been applied upstream.

File: games-action/descent1-freedata/descent1-freedata-1.ebuild (hunk 1 above)
[the remark this answers is not in the document]
Comment (dr-diem) [reply games-action/descent1-freedata/descent1-freedata-1.ebuild:38]: Applied upstream.

File: games-action/dxx-rebirth/metadata.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (dr-diem) [reply games-action/dxx-rebirth/metadata.xml:10]: Currently testing locally. When that's done I'll tackle the rebase and combining the commits to just one, that has the correct headers including sign-off.

[the remark this answers is not in the document]
Comment (dr-diem) [reply games-action/dxx-rebirth/metadata.xml:10]: Okay, the changes are all in and the commit history cleaned up. I did rebase to latest master, so I don't know why there remains a conflict on metadata.xml. Feel free to remove me as moderator; I am not listed in the file in my repo.

The only other CI warnings are for the copyright message (which we've discussed) and now from the 0.59 ebuild that doesn't recognize the USE flags that the new ebuild introduces.

Looking forward to your feedback @chewi !

**Comment by dr-diem** [comment]
I'm withdrawing this PR on the basis that I have combined changes relating to three, albeit inter-related, packages in one commit. I will raise a new one containing three separate 'clean' commits.

**Comment by chewi** [comment]
As discussed, remember you can always `git push --force` over any existing PR with a completely new set of changed.

**Pull request closed without merging**